### PR TITLE
Documentation of Parallel Configuration Parameters and Fixes to Defect Vertex Perfect Matching in ParallelSolver

### PR DESCRIPTION
**Pull request by YangLiuWillow**
- Added python script and a markdown file on the parallel configuration using the fusion_blossom library 
- Fixed the defect vertex perfect matching "our of index range" issue in ParallelSolver by adding the 'defect_perfect_matching' method to SolverParallel struct in Rust. 
- Attempted to note down the function input and output for the fusion_blossom library.

### CODE REVIEW

**Comment by YangLiuWillow** [comment]
Fixes #34

File: src/util.rs [hunk 1]
```diff
@@ -697,6 +700,7 @@ pub struct BenchmarkProfiler {
 impl BenchmarkProfiler {
     pub fn new(noisy_measurements: VertexNum, detail_log_file: Option<(String, &PartitionInfo)>) -> Self {
         let benchmark_profiler_output = detail_log_file.map(|(filename, partition_info)| {
+            println!("here the filename is {}", filename);
```
Comment (yuewuo) [inline src/util.rs:703]: Consider removing unnecessary prints

File: tutorial/src/demo/example-parallel-yl.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
+# Example Parallel Configuration 
+
+In this chapter, you will learn how about the configuration of graph partitions and different fusion plans as explained in the paper [1]. Make sure you understand the decoding and syndrome graph before proceeding. You can [download the complete code here](./example-parallel-yl.py).
+
+In paper [1], Fusion Blossom solves the MWPM problem with a Divide and Conquer approach. Fusion Blossom *divides* the decoding problem into two sub-problems that can be solved, or "conquered," independently and *fuses* their solutions recursively to obtain the overall solution. This recursive division/fusion is represented as a full binary tree, denoted as a *fusion tree*.
```
Comment (yuewuo) [inline tutorial/src/demo/example-parallel-yl.md:5]:
```suggestion
In paper [1], Fusion Blossom solves the MWPM problem with a Divide and Conquer approach. Fusion Blossom *divides* the decoding problem into two sub-problems that can be solved, or "conquered", independently and *fuses* their solutions recursively to obtain the overall solution. This recursive division/fusion is represented as a full binary tree, denoted as a *fusion tree*. 
```

File: tutorial/src/demo/example-parallel-yl.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
+# Example Parallel Configuration 
+
+In this chapter, you will learn how about the configuration of graph partitions and different fusion plans as explained in the paper [1]. Make sure you understand the decoding and syndrome graph before proceeding. You can [download the complete code here](./example-parallel-yl.py).
```
Comment (yuewuo) [inline tutorial/src/demo/example-parallel-yl.md:3]: consider removing the `-yl` suffix

File: tutorial/src/demo/example-parallel-yl.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
+import fusion_blossom as fb
```
Comment (yuewuo) [inline tutorial/src/demo/example-parallel-yl.py:1]: remove the filename to remove `-yl` suffix

File: tutorial/src/demo/python-documentation.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
+# Python Library Documentation 
+
+* `CircuitLevelPlanarCode`
+* `CodeCapacityPlanarCode(d=int, p=float, max_half_weight=int)`
+    * `d`: code distance
+    * `p`: physical error rate
+    * `max_half_weight`: the library scales the edge weight such that the maximum weight is 500 * 2
```
Comment (yuewuo) [inline tutorial/src/demo/python-documentation.md:7]:
```suggestion
    * `max_half_weight`: the library scales the edge weight such that the maximum weight is 2 * `max_half_weight`
```

**Review by yuewuo — CHANGES_REQUESTED** [review]

**Review by YangLiuWillow — COMMENTED** [review]
I just modified the files according to the review.

**Review by yuewuo — APPROVED** [review]

**Pull request merged**